### PR TITLE
Pipeline Launch: dynamic parameters

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes

Display dynamic pipeline parameters for pipelines that require them (e.g. MentaList).  These are parameters that are pull from Galaxy (e.g. which database to use).

## Related issue

Related to #777 

## Checklist

Things for the developer to confirm they've done before the PR should be accepted:

*   [ ] ~CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~
*   [ ] ~Tests added (or description of how to test) for any new features.~
*   [ ] ~User documentation updated for UI or technical changes.~

### CODE REVIEW

File: src/main/webapp/resources/js/pages/launch/LaunchForm.jsx [hunk 1]
```diff
@@ -40,51 +39,26 @@ export function LaunchForm() {
...
+            <LaunchParameters form={form} />
             <SharePipelineResults />
             <ReferenceFiles />
-            <SavedParameters form={form} />
-            <ParameterWithOptions parameters={state.parameterWithOptions} />
             <LaunchFiles />
           </Space>
         </Col>
-        <Col sm={24} md={5}>
-          <Anchor>
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/launch/LaunchForm.jsx:54]: Just a question. Why did you get rid of the anchors? I liked how it looked and the functionality made sense

File: src/main/webapp/resources/js/pages/launch/ParameterWithOptions.jsx [hunk 2]
```diff
@@ -15,7 +16,9 @@ import { Checkbox, Form, Radio, Select } from "antd";
  * @returns {*}
  * @constructor
  */
-export function ParameterWithOptions({ parameters }) {
+export function ParameterWithOptions() {
+  const [{ parameterWithOptions: parameters }] = useLaunch();
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/launch/ParameterWithOptions.jsx:20]: Another question. What is this? I haven't seen this before (the format)

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Did a quick review but looking good overall. Just a few changes and questions.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java [hunk 3]
```diff
@@ -128,19 +138,60 @@ public UIPipelineDetailsResponse getPipelineDetails(UUID id, Locale locale) thro
...
          */
 		detailsResponse.setAcceptsSingleSequenceFiles(description.acceptsSingleSequenceFiles());
 		detailsResponse.setAcceptsPairedSequenceFiles(description.acceptsPairedSequenceFiles());
 
-        return detailsResponse;
-    }
+		/*
+		Dynamic Sources
+		 */
+		if (description.requiresDynamicSource()) {
```
Comment (deepsidhu85) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java:155]: Please add comments to this block of code to clarify what's happening

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]

File: src/main/webapp/resources/js/pages/launch/LaunchForm.jsx (hunk 1 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/LaunchForm.jsx:54]: As per our discussion on Friday, they were of now real benifit.

File: src/main/webapp/resources/js/pages/launch/ParameterWithOptions.jsx (hunk 2 above)
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/ParameterWithOptions.jsx:20]: This is [destructing with renaming](https://wesbos.com/destructuring-renaming).  Essentially shorthand for:

```
const [state] = useLaunch(); // Array destructure;
const {parameterWithOptions : parameters} = state; // Object destructure with renaming to paramters
```

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java (hunk 3 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java:155]: Tried updating in a7628b5

File: src/main/webapp/resources/js/pages/launch/LaunchForm.jsx [hunk 4]
```diff
@@ -40,51 +39,26 @@ export function LaunchForm() {
...
-                  icon={<IconLaunchPipeline />}
-                >
-                  {i18n("LaunchContent.submit")}
-                </Button>
-              }
-            />
-          </Anchor>
+        <Col sm={24} md={6}>
+          <Button
+            style={{ position: "sticky", top: 20 }}
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/LaunchForm.jsx]: Updated in one of those pesky commits ;)

File: src/main/webapp/resources/js/pages/launch/DynamicSources.jsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import React from "react";
+import { Form, Select } from "antd";
+import { useLaunch } from "./launch-context";
+
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply src/main/webapp/resources/js/pages/launch/DynamicSources.jsx:4]: Updated in e799d40

File: src/main/webapp/resources/js/pages/launch/ParameterWithOptions.jsx (hunk 2 above)
Comment (deepsidhu85) [reply src/main/webapp/resources/js/pages/launch/ParameterWithOptions.jsx:20]: Awesome. Thanks!

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java (hunk 3 above)
Comment (deepsidhu85) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPipelineService.java:155]: Very nicely done

File: src/main/webapp/resources/js/pages/launch/LaunchContent.jsx [hunk 6]
```diff
@@ -12,10 +13,33 @@ import { SPACE_LG } from "../../styles/spacing";
 export function LaunchContent() {
   const [{ pipeline }] = useLaunch();
 
+  const [validCart, setValidCart] = React.useState(true);
+
+  /**
+   * Listen for an empty cart, if empty, DO WHAT???
```
Comment (deepsidhu85) [inline src/main/webapp/resources/js/pages/launch/LaunchContent.jsx:19]: Update this comment

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]

**Review by deepsidhu85 — APPROVED** [review]

**Pull request merged**